Jenkinsfile

```diff
@@ -0,0 +1,257 @@
+            ]
+        ]
+     ])
+}
+
+
+def unittest(String pythonVersion) {
+    echo "set ${pythonVersion} github status"
+    String buildName = "linux ${pythonVersion} test"
+    setBuildStatus('PENDING', "${buildName}", 'test start')
+    try {
+        if (pythonVersion == "py3") {
+            // create python 3 venv
+            sh """
+            python3 -m venv .py3venv --without-pip
+            . .py3venv/bin/activate
+            curl https://bootstrap.pypa.io/get-pip.py | python
+            id
+            python setup.py install
+            pip install coverage mock
+            pip install -I mbed-ls==1.2.14
+            coverage run -m unittest discover -s test -vvv
+            coverage html
+            coverage xml
+            deactivate
+            """
+        } else {
+            // create python2 venv, do installation and run tests
+            sh """
+            virtualenv --python=../usr/bin/python .py2venv --no-site-packages
+            . .py2venv/bin/activate
+            id
+            python setup.py install
+            pip install coverage mock
+            coverage run -m unittest discover -s test -vvv
+            coverage html
+            coverage xml
+            deactivate
+            """
+        }
+        setBuildStatus('SUCCESS', "${buildName}", "test done")
+    } catch (err) {
+        setBuildStatus('FAILURE', "${buildName}", "test failed")
+        currentBuild.result = 'FAILURE'
+    }
+}
+
+
+def linux_pylint_check() {
+    // execute pylint check shell (only for python2, and linux)
+    sh './run_pylint.sh | tee pylint.log'
+
+    // publish warnings
+    step([
+        $class: 'WarningsPublisher',
+        consoleParsers: [
+            [
+                parserName: 'GNU Make + GNU C Compiler (gcc)'
+            ]
+        ],
+        parserConfigurations: [
+            [
+                parserName: 'PyLint',
+                pattern: '**/pylint.log'
+            ]
+        ],
+        unstableTotalAll: '0',
+        failedTotalAll: '0'
+    ])
+}
+
+
+def winTest(String pythonVersion) {
+    echo "set ${pythonVersion} github status"
+    String buildName = "windows ${pythonVersion} test"
+    setBuildStatus('PENDING', "${buildName}", 'test start')
+    try {
+        if (pythonVersion == "py3") {
+            // create python 3 venv
+            bat 'cd \\'
+            bat 'c:\\Python36\\python -m venv c:\\Jenkins\\ws\\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\\mbed-flasher\\.py3venv'
+            bat 'cd c:\\Jenkins\\ws\\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\\mbed-flasher'
+            bat """
+            .py3venv\\Scripts\\activate
+            python setup.py install
+            pip install coverage mock
+            pip install -I mbed-ls==1.2.14
+            coverage run -m unittest discover -s test -vvv
+            coverage html
+            coverage xml
+            deactivate
+            """
+        } else {
+            // create python2 venv, do installation and run tests
```

Comment:
for py2,  after activate, it's just jump to sth else, instead of installing
```
11:51:51 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>virtualenv py2venv --no-site-packages 
11:52:17 [windows] New python executable in C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher\py2venv\Scripts\python.exe
11:52:47 [windows] Installing setuptools, pip, wheel...done.
11:52:47 [windows] 
11:52:47 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>py2venv\Scripts\activate
[Pipeline] [windows] step
11:52:47 [windows] [Set GitHub commit status (universal)] SUCCESS on repos [GHRepository@34cff795[description=Test Framework for Command Line interface,h
```

Reply:
okay, the absolute path has been fixed, but just after activate the venv, jobs suddenly stopped. I will look more tmr.